segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerMetadataTests.java

```diff
@@ -263,8 +272,10 @@ public void testDeleteStreamSegment() {
             deletedStreamSegmentIds.add(segmentId);
             for (int transIndex = 0; transIndex < TRANSACTIONS_PER_SEGMENT_COUNT; transIndex++) {
                 long transactionId = segmentIds.get(index + transIndex + 1);
-                deletedStreamSegmentIds.add(transactionId);
-                expectedDeletedSegmentNames.add(m.getStreamSegmentMetadata(transactionId).getName());
+                if (deletedStreamSegmentIds.add(transactionId)) {
```

Comment:
This change as is does not change anything in the flow of the test case. Without the if, the call to add to the hash set would simply not add a copy of the element if it is already there (it is a set after all). Perhaps what we should do instead is to actually verify that the segments we expect to be added are there. What do you think?

Reply:
That is done below. Check out line 281:
```
            AssertExtensions.assertContainsSameElements("Unexpected StreamSegments were deleted.", expectedDeletedSegmentNames, deletedSegmentNames);
```

Reply:
Ok.